Ports <Dmicon> and adds wall closets

BarteG44 opened the pull request:
## About The Pull Request

Ports https://github.com/Monkestation/Monkestation2.0/pull/3792 from monke and adds wall closets

## Why It's Good For The Game

\<Dmicon> is a better way of getting icons into TGUI than icon2base64 (as far as i know) and is necessary for the closets to work as envisioned. The closets themselves add just a bit more flavor to the stations, filling out the space on empty walls and allowing you to store items more neatly and compactly 

This PR is also required for #12030

## Testing Photographs and Procedure

![image](https://github.com/user-attachments/assets/1f7cc92a-e011-4349-93ed-9e915468e233)

![image](https://github.com/user-attachments/assets/9a871b38-e155-4493-9284-eb5a02dae2a6)

![image](https://github.com/user-attachments/assets/9dc66d88-8458-42fe-829b-3584a00c4cae)

https://github.com/user-attachments/assets/36d0e27f-51e0-46d5-9573-62397408a43c


</details>

## Changelog
:cl: BarteG44, PowerfulBacon, Absolucy, Aylong, jlsnow301
add: Added wall closets
code: Ported \<Dmicon>
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

BarteG44 committed:
Ports <Dmicon> and adds wall closets

BarteG44 committed:
messed up while copy pasting

BarteG44 committed:
Technically works

BarteG44 committed:
idk man i don't get this

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into port-of-a-port-of-a-port-of-dmicon

BarteG44 committed:
i guess

BarteG44 committed:
lints?

also made the edge cases work with cringe snowflake code and icon2base64. Hopefully someone lets me know of a better solution

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into port-of-a-port-of-a-port-of-dmicon

BarteG44 committed:
prettier also screwdrivers work now but are jank

BarteG44 committed:
wider conditions

BarteG44 committed:
no more icon2base64

thanks absolucy

BarteG44 committed:
updates the icons AFTER initialize

BarteG44 committed:
adds a command closet

BarteG44 committed:
basically a typo

BarteG44 committed:
you can now take them apart. revolutionary

itsmeow commented on code/modules/asset_cache/asset_list_items.dm:
```diff
@@ -640,3 +640,32 @@
+	name = "icon_ref_map"
+	early = TRUE
+
+/datum/asset/json/icon_ref_map/generate()
```
(non blocking) this proc is extremely slow btw, it takes around 4 seconds to fully generate

Might be worth offloading walking the file tree to rustg since it's such a simple operation

BarteG44 replied:
This is part of the Ctrl+C Ctrl+V "port" so i'm not even sure what it does. I also don't really know what rustg is so i'm gonna need help with that

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into port-of-a-port-of-a-port-of-dmicon

BarteG44 committed:
Adds sounds to closet interactions

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into port-of-a-port-of-a-port-of-dmicon

BarteG44 committed:
forgot an isliving check for sound

BarteG44 committed:
adds closets to all maps

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into port-of-a-port-of-a-port-of-dmicon

BarteG44 committed:
meata

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into port-of-a-port-of-a-port-of-dmicon

BarteG44 committed:
small changes

BarteG44 committed:
fixes icons being blurry on 516

BarteG44 committed:
Allows borgs to look into the lockers, but not to interact with them.

Also stealth fixes the teleporter UI not auto updating, causing the calibration status not being obvious

BarteG44 committed:
fixes some mapping issues

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into port-of-a-port-of-a-port-of-dmicon

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into port-of-a-port-of-a-port-of-dmicon

BarteG44 committed:
removes changes to maps

PowerfulBacon commented on code/game/objects/structures/crates_lockers/wall_closets.dm:
```diff
@@ -0,0 +1,285 @@
+	if(contents.len >= 20)
+		return FALSE
+	if(!ui_index)
+		for(var/index = 1, index <= closet_contents.len, index++)
```
Use ranged for loops. This review applies to all for loops in this PR using the C style.
```
for(var/i in 1 to 20)
```

BarteG44 replied:
addressed (i believe)

PowerfulBacon reviewed (changes requested):
Follow appropriate DM naming standards. DM uses snake_case for procs and they should not start with a capital letter.

Lines should also not end with a `;` in dm.

Full code standards are listed at https://github.com/BeeStation/BeeStation-Hornet/wiki/Code-Standards

BarteG44 committed:
fixes procs and for loops

BarteG44 committed:
removes the semicolons

why were they even there

Tsar-Salat reviewed (changes requested):
Its my opinion that the wall closets have too wide(++) and slightly too tall(+) a profile. I would like to see their overall size on the wall reduced to around where most typical wallmachines are. A discussion with pigeon a while back had them of similiar opinion.

Less than a poster, more than a firealarm

The application of DMIicon I am less familiar with, but I saw no reports of issues during the testmerge, so I have no issue approving/merging when the size and other maints reviews are addressed.

BarteG44 committed:
puts the lockers into the shrink-inator

BarteG44 committed:
this offset looks better

BarteG44 committed:
fixes closet proximity checking and adds an open state sprite

BarteG44 committed:
adjusts the sprites

BarteG44 committed:
fixed multiple people logic
